test/evergreen/run_format_configs.sh

```diff
@@ -21,20 +20,103 @@ fi
 
 success=0
 failure=0
+running=0
+parallel_jobs=8
+cleanup=0
+PID=""
+declare -a PID_LIST
+
+usage() {
+	echo "usage: $0 "
+	echo "    -j parallel  jobs to execute in parallel (defaults to 8)"
+
+	exit 1
+}
+
+
+while :; do
+	case "$1" in
+	-j)
+		parallel_jobs="$2"
+		[[ "$parallel_jobs" =~ ^[1-9][0-9]*$ ]] ||
+			fatal_msg "-j option argument must be a non-zero integer"
+		shift ; shift ;;
+	-*)
+		usage ;;
+	*)
+		break ;;
+	esac
+done
+
+# Wait for other format runs.
+wait_for_process()
+{
+	while true
+	do
+		for process in ${PID_LIST[@]};do
+			if [ ps $process > /dev/null ] ; then
+				# The process id is running so sleep for 2 second before checking another
+				# process status.
+				sleep 2
+			else
+				# Remove the process id of the completed runs from the process id array.
+				PID_LIST=(${PID_LIST[@]/$process})
+
+				#wait for the process to get the exit status.
+				wait $process
+				exit_status=$?
 
+				let "running--"
+
+				config_name=`egrep $process $tmp_file | awk -F ":" '{print $2}' | rev | awk -F "/" '{print $1}' | rev`
+				if [ $exit_status -ne "0" ]; then
+					let "failure++"
+					[ -f WT_TEST_${config_name}/CONFIG ] && cat WT_TEST_${config_name}/CONFIG
+				else
+					let "success++"
+					# Remove database files of successful jobs.
+					[ -d WT_TEST_${config_name} ] && rm -rf WT_TEST_${config_name}
+				fi
+
+				echo "Exit status of config ${config_name} is ${exit_status}"
+				break
```

Comment:
I'm wondering if `continue` makes more sense here to enable quickly checking other processes and moving those finished out of the list.

Reply:
Fixed

Reply:
Suggest calling `continue` explicitly.